src/main/webapp/i18n/en/exercise.json

```diff
@@ -165,24 +165,29 @@
             },
             "form": {
                 "title": {
+                    "undefined": "There has to be a title set!",
                     "pattern": "The title must only consist of alphanumeric characters and '_', '-' or whitespaces!"
                 },
                 "shortName": {
+                    "undefined": "There has to be a short name set!",
                     "pattern": "The short name must begin with a letter and must not contain any special characters!",
                     "minlength": "The short name must be at least 3 characters long!"
                 },
                 "packageName": {
+                    "undefined": "There has to be a package name set!",
                     "pattern": {
                         "JAVA": "The package name must consist of one or more valid Java identifiers separated by '.', e.g. \"net.java\"!",
                         "KOTLIN": "The package name must consist of one or more valid Kotlin identifiers separated by '.', e.g. \"net.kotlin\"!",
                         "SWIFT": "The package name must consist of one or more valid Swift identifiers without any separator, e.g. \"SwiftEx\"!"
                     }
                 },
                 "points": {
+                    "undefined": "There has to be a value for the points set.",
                     "customMin": "The number of points must be at least 1.",
                     "customMax": "The number of points may not exceed 9999."
                 },
                 "bonusPoints": {
+                    "undefined": "There has to be a value for the bonus points set.",
```

Comment:
Agree on exlcamation marks, changed in bb2dd96.

Reply:
> 2. **Maybe a same-sized badge as those buttons and some margins wouldn't hurt the eye and it would look much ore beautiful.** Don't know though how things here exactly work and if this is in the scope of the ticket.
>    ![different_size](https://user-images.githubusercontent.com/94109535/142909004-5757a807-6bf6-4b9b-9c8b-fa1da9c2b723.PNG)

This badge is used also for other components. I think it would be out of scope to change these. As well as the same size might suggest that it is a button.